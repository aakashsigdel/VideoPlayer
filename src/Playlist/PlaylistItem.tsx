import * as React from 'react'
import cn from 'classnames'

import { Play, Pause } from '../icons'
import { IVideo } from '../VideoPlayer'
import styles from './Playlist.module.css'

export interface IPlaylistItemProps {
  video: IVideo
  videoNumber: number
  onSelectVideo: (videoNumber: number) => void
  nowPlaying: boolean
}

export const PlaylistItem: React.FunctionComponent<IPlaylistItemProps> = ({
  video,
  videoNumber,
  onSelectVideo,
  nowPlaying,
}) => (
  <li
    onClick={() => onSelectVideo(videoNumber)}
    className={cn(styles.listItem, { [styles.nowPlaying]: nowPlaying })}
  >
    <span className={styles.number}>{videoNumber + 1}</span>
    <span className={cn(styles.title, styles.ellipsis)}>{video.title}</span>
    <span className={cn(styles.artist, styles.ellipsis)}>{video.artist}</span>
  </li>
)
