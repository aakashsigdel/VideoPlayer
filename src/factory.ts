export const getSong = (suffix: string = '1') => ({
  title: `testSong${suffix}`,
  artist: `testArtist${suffix}`,
  videoUrl: `testUrl${suffix}`,
})

export const getSongs = () => {
  return ['1', '2', '3'].map(getSong)
}
