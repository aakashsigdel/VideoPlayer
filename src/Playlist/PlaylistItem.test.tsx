import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { create } from 'react-test-renderer'

import { PlaylistItem, IPlaylistItemProps } from './PlaylistItem'
import { getVideo } from '../factory'

describe('<PlaylistItem />', () => {
  let wrapper: ShallowWrapper<IPlaylistItemProps>
  const defaultProps: IPlaylistItemProps = {
    video: getVideo(),
    videoNumber: 0,
    onSelectVideo: jest.fn(),
  }
  beforeEach(() => {
    wrapper = shallow(<PlaylistItem {...defaultProps} />)
  })
  it('should render correctly', () => {
    const tree = create(<PlaylistItem {...defaultProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call onSelectvideo onClick', () => {
    wrapper.find('li').simulate('click')
    expect(defaultProps.onSelectVideo).toHaveBeenCalledWith(
      defaultProps.videoNumber
    )
  })
})
