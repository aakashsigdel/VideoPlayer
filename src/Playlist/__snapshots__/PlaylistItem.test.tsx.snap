// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlaylistItem /> should render correctly 1`] = `
<li
  onClick={[Function]}
>
  testVideo0
</li>
`;
